tools/macos/docker-machine/README.md

```diff
@@ -0,0 +1,124 @@
+# Setting up OpenWhisk with Docker-machine
+
+OpenWhisk will be running on virtual machine in which Docker daemon is running.
+You will make provision of a virtual machine with Docker-machine and communicate with them via Docker remote API.
+
+# Prerequisites
+
+The following are required to build and deploy OpenWhisk from a Mac host:
+
+- [Oracle VM VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+- [Docker 1.12.0](https://docs.docker.com/engine/installation/mac/) (including `docker-machine`)
+- [Java 8](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+- [Scala 2.11](http://scala-lang.org/download/)
+- [Ansible 2.1.2.0](http://docs.ansible.com/ansible/intro_installation.html)
+
+**Tip** Versions of Docker and Ansible are lower than the latest released versions, the versions used in OpenWhisk are pinned to have stability during continuous integration and deployment.
+
+
+[Homebrew](http://brew.sh/) is an easy way to install all of these and prepare your Mac to build and deploy OpenWhik. The following shell command is provided for your convenience to install `brew` with [Cask](https://github.com/caskroom/homebrew-cask) and bootstraps these to complete the setup. Copy the entire section below and paste it into your terminal to run it.
+
+```
+echo '
+# install homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+# install cask
+brew tap caskroom/cask
+# install virtualbox
+brew cask install virtualbox
+# install docker 1.12.0
+brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/33301827c3d770bfd49f0e50d84e0b125b06b0b7/Formula/docker.rb
+# install docker-machine
+brew install docker-machine
+# install java 8
+brew cask install java
+# install scala
+brew install scala
+# install pip
+sudo easy_install pip
```

Comment:
See comments above about using: `brew install --upgrade python` and about removing `sudo` from pip installs.